src/controller/api/newsAPIController.js

```diff
@@ -6,39 +6,22 @@ const getDataFromCheerio = require('../../utils/getDataFromCheerio');
 const getSourceImageFromCheerio = require('../../utils/getSourceImageFromCheerio');
 const pagination = require('../../utils/paginate');
 const truncation = require('../../utils/truncate');
+const { getArticles } = require('../../utils/getArticles');
 const defaultImage =
   'https://raw.githubusercontent.com/MizouziE/energy-prices-api/master/public/img/energy-prices-api-socials.png';
-let articles = {};
-
-sources.forEach(async (source) => {
-  try {
-    const response = await axios.get(source.site);
-    const returnedArticles = getDataFromCheerio(response.data);
 
-    returnedArticles.forEach(async (returned) => {
-      // now we have an array of multiple article's data to iterate through
-      articles[
-        new sc(returned.url)
-          .encryptToQRCodeAlphanumeric(returned.url)
-          .slice(0, 10)
-      ] = {
-        // use short-crypt to "hash" the url which works as unique ID
-        title: returned.title,
-        url: source.base + returned.url,
-        source: source.name,
-        image: /^((http|https):\/\/)/.test(returned.image)
-          ? returned.image
-          : defaultImage,
-        region: source.region,
-      };
-    });
-  } catch (error) {
-    console.log({ error });
-    return;
-  }
-});
+let articles = {};
 
+async function fetchArticles() {
+  articles = await getArticles();
+}
+fetchArticles();
 
+// fetch every 30 minutes
+setInterval(() => {
+  console.log("updating catalogue...");
+  fetchArticles();
+}, 1800000);
```

Comment:
```suggestion
}, 1000 * 60 * 30);
```

I think this is more clear :sparkles: